shopify/usage-subscription-template/api/models/shopifyShop/actions/subscribe.ts

```diff
@@ -3,25 +3,40 @@ import {
   preventCrossShopDataAccess,
   save,
   ActionOptions,
-  SubscribeShopifyShopActionContext,
 } from "gadget-server";
-import { trialCalculations, convertCurrency } from "../../../../utilities";
-
-/**
- * @param { SubscribeShopifyShopActionContext } context
- */
-export async function run({
+import { trialCalculations } from "../../../../utilities";
+
+export type SubscriptionLineItems = {
+  id: string;
+  plan: {
+    pricingDetails: {
+      __typename: "AppUsagePricing";
+      cappedAmount: {
+        amount: string;
+        currencyCode: string;
+      };
+      balanceUsed: {
+        amount: string;
+        currencyCode: string;
+      };
+      interval: string;
+      terms: string;
+    };
+  };
+}[];
+
+export const run: ActionRun = async ({
   params,
   record,
   logger,
   api,
   connections,
   currentAppUrl,
-}) {
+}) => {
   applyParams(params, record);
   await preventCrossShopDataAccess(params, record);
 
-  const planMatch = await api.plan.maybeFindOne(params.planId, {
+  const planMatch = await api.plan.maybeFindOne(params.planId as string, {
```

Comment:
same comment - we don't have to typecast if we check for falsy, which is a pattern we should probably follow (but I can actually provide a diff in the PR this time)
```suggestion
 if(!params.planId) {
  throw new Error("No planId provided, subscription cannot be created);
 }

  const planMatch = await api.plan.maybeFindOne(params.planId, {
```